homeassistant/components/thermostat/zwave.py

```diff
@@ -116,6 +116,8 @@ def update_properties(self):
             temps.append(int(value.data))
             if value.index == self._index:
                 self._target_temperature = value.data
+                if self._unit is None:
```

Comment:
You mean defaulting the whole system unit? 
If your units are F and Danfoss is C, then it will show wrong temps.